iml-agent/src/daemon_plugins/network.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+//! #Network daemon-plugin
+//!
+//! This module is responsible for continually fetching the network interfaces and their respective stats.
+//!
+//!
+
+use crate::{
+    agent_error::ImlAgentError,
+    daemon_plugins::{DaemonPlugin, Output},
+    network_interfaces,
+};
+use futures::Future;
+use iml_wire_types::NetworkInterface;
+use std::pin::Pin;
+
+#[derive(Debug)]
+pub struct Network;
+
+pub fn create() -> impl DaemonPlugin {
+    Network
+}
+
+async fn get_network_interfaces<F1>(get_interfaces: fn() -> F1) -> Result<Output, ImlAgentError>
```

Comment:
What's the benefit of having this fn?

Reply:
I did this initially because I thought I was going to add a unit test but it turns out I don't really need one here. I'll get rid of the function entirely if you feel that there is too much abstraction.

Reply:
Removed